pallets/protos/src/tests.rs

```diff
@@ -1,385 +1,851 @@
-			Origin::signed(sp_core::ed25519::Public::from_raw(PUBLIC1)),
-			references,
-			Categories::Text(TextCategories::Plain),
-			Vec::new(),
-			None,
-			None,
-			data,
-		));
-
-		assert!(<Protos<Test>>::contains_key(PROTO_HASH));
-	});
-}
+			_ = upload(dd.account_id, &stake.proto_fragment);
 
-#[test]
-fn upload_should_not_works_if_proto_hash_exists() {
-	new_test_ext().execute_with(|| {
-		let data = DATA.as_bytes().to_vec();
-		initial_upload();
-		let references = vec![];
-
-		assert_noop!(
-			ProtosPallet::upload(
-				Origin::signed(sp_core::ed25519::Public::from_raw(PUBLIC1)),
-				references,
-				Categories::Text(TextCategories::Plain),
-				Vec::new(),
-				None,
-				None,
-				data,
-			),
-			Error::<Test>::ProtoExists
-		);
-	});
-}
+			_ = lock_(&stake.lock);
+			_ = link_(&stake.lock.link);
 
-#[test]
-fn patch_should_works() {
-	new_test_ext().execute_with(|| {
-		let pair = sp_core::ecdsa::Pair::from_string("//Alice", None).unwrap();
-		let immutable_data = DATA.as_bytes().to_vec();
-		initial_upload();
-
-		let data = immutable_data.clone();
-		let proto_hash = blake2_256(&immutable_data);
-
-		assert_ok!(ProtosPallet::patch(
-			Origin::signed(sp_core::ed25519::Public::from_raw(PUBLIC1)),
-			proto_hash,
-			Some(Compact(123)),
-			vec![],
-			None,
-			data,
-		));
-
-		assert_eq!(<Protos<Test>>::get(proto_hash).unwrap().include_cost, Some(Compact(123)))
-	});
-}
+			_ = stake_(
+				stake.lock.link.clamor_account_id,
+				&stake.proto_fragment,
+				&stake.get_stake_amount(),
+			);
 
-#[test]
-fn patch_proto_should_not_work_if_user_is_unauthorized() {
-	new_test_ext().execute_with(|| {
-		let data = DATA.as_bytes().to_vec();
-		initial_upload();
-
-		let (pair, _) = sp_core::ed25519::Pair::generate();
-
-		assert_noop!(
-			ProtosPallet::patch(
-				Origin::signed(pair.public()),
-				PROTO_HASH,
-				Some(Compact(123)),
-				vec![],
-				None,
-				data,
-			),
-			Error::<Test>::Unauthorized
-		);
-	});
-}
+			let proto_with_refs = ProtoFragment {
+				references: vec![stake.proto_fragment.get_proto_hash()],
+				..dd.proto_fragment
+			};
 
-#[test]
-fn patch_proto_should_not_work_if_proto_not_found() {
-	new_test_ext().execute_with(|| {
-		let immutable_data = DATA.as_bytes().to_vec();
-		let proto_hash = blake2_256(immutable_data.as_slice());
-		assert_noop!(
-			ProtosPallet::patch(
-				Origin::signed(sp_core::ed25519::Public::from_raw(PUBLIC1)),
-				proto_hash,
-				Some(Compact(123)),
-				vec![],
-				None,
-				immutable_data,
-			),
-			Error::<Test>::ProtoNotFound
-		);
-	});
-}
+			assert_ok!(upload(stake.lock.link.clamor_account_id, &proto_with_refs));
+		});
+	}
+
+	#[test]
+	fn upload_should_not_work_if_user_did_not_stake_enough() {
+		new_test_ext().execute_with(|| {
+			let dd = DummyData::new();
+
+			let stake = dd.stake;
 
-#[test]
-fn patch_should_not_work_if_detached() {
-	let keystore = KeyStore::new();
-	let mut t = new_test_ext();
+			_ = upload(dd.account_id, &stake.proto_fragment);
```

Comment:
we are ignoring a lot of results here...

Reply:
Since `upload()` is tested. We can safely assume that if all the `upload()` tests worked, then this should also return `Ok()`?

Reply:
makes sense. I'll do `unwrap` instead of `assert_ok!`.

othherwise, there'll be a lot of `assert_ok`s 😆

Reply:
so we won't know what we're testing